bigquery_etl/cli/metadata.py

```diff
@@ -111,23 +119,36 @@ def update(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None
     ConfigLoader.get("default", "project", fallback="moz-fx-data-shared-prod")
 )
 @sql_dir_option
-def publish(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None:
+@parallelism_option()
+def publish(
+    name: str, sql_dir: Optional[str], project_id: Optional[str], parallelism: int
+) -> None:
     """Publish Bigquery metadata."""
-    client = bigquery.Client(project_id)
+    credentials = get_credentials()
```

Comment:
Re-using the credentials here made a significant difference (from 4 minutes to about 1 minute run time, at least for the subset I tested this on).